app/models/concerns/orchestration/dhcp.rb

```diff
@@ -87,13 +86,21 @@ def boot_server
     bs = tftp.bootServer
     # if that failed, trying to guess out tftp next server based on the smart proxy hostname
     bs ||= URI.parse(subnet.tftp.url).host
-    # now convert it into an ip address (see http://theforeman.org/issues/show/1381)
-    ip = NicIpResolver.new(:nic => self).to_ip_address(bs) if bs.present?
-    return ip unless ip.nil?
 
-    failure _("Unable to determine the host's boot server. The DHCP smart proxy failed to provide this information and this subnet is not provided with TFTP services.")
-  rescue => e
-    failure _("failed to detect boot server: %s") % e, e
+    # only smart proxies with V2 API are supported
+    if subnet.dhcp.has_capability?(:DHCP, :dhcp_filename_hostname)
+      # we no longer convert the boot server to IPv4 - smart proxy modules are required to do so if they don't support hostname
+      return bs
+    else
```

Comment:
Technically the whole indenting in else here is not needed since the `if` statement returns. That makes the diff easier to read.

Reply:
My reasoning is mostly that the diff is smaller so it's easier to see the actual differences vs some whitespace changes, but I won't block on it. Just a hint for the future.